### PR TITLE
[GT-145] Fixing, Refreshing and Enriching the Service Group Sample Data

### LINKED ISSUE #416: Refresh and enrich ServiceGroups from sample data
Follows on from #394 

- [ ] Extend `lib/Doctrine/deploy/AddServiceGroups.php` so that it can create ServiceGroups that actually contain Services
- [ ] Add `lib/Doctrine/deploy/AddServiceGroupRoles.php` so that ServiceGroup level roles can be created.
- [ ] Refresh `lib/Doctrine/deploy/sampleData/ServiceGroups.xml` and `/lib/Doctrine/deploy/sampleData/ServiceGroupRoles.xml` to be consistent with the rest of the sample data.

The best way to tackle this may be to deploy the existing sample data to an instance, create some ServiceGroups in the UI, then use the [API output](https://gocdb.github.io/api/read/get_service_group/) to update the `ServiceGroups.xml`. Because the GOCDB UI doesn't let you request a role on behalf of a user, `ServiceGroupRoles.xml` can start from the [API output](https://gocdb.github.io/api/read/get_service_group_role/) but then it will have to be manually updated to include user roles.

### PR DESCRIPTION
**Pull request by rowan04**
will resolve #416 when finished

### CODE REVIEW

**Comment by rowan04** [comment]
rebased to remove extra commit

**Comment by rowan04** [comment]
to note - all service groups are currently incorrectly given only local scope

**Comment by rowan04** [comment]
> to note - all service groups are currently incorrectly given only local scope

fixed :)

**Comment by rowan04** [comment]
Codacy and codeclimate issues to be fixed. Other than that, ready for review

**Comment by rowan04** [comment]
force push and rebase to merge 3 commits on code formatting fixes into 1 commit.
All codacy and codeclimate issues are now fixed.

File: lib/Doctrine/deploy/AddServiceGroupRoles.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+         * one role type
+         */
+        if (count($roleTypes) !== 1) {
+            throw new Exception(count($roleTypes) . " role types " .
+                "found with name: " . $userRole);
+        }
+
+        foreach ($roleTypes as $result) {
+            $roleType = $result;
+        }
```
Comment (gregcorbett) [inline lib/Doctrine/deploy/AddServiceGroupRoles.php:40]: What is this (and similar) for loop doing?

**Review by gregcorbett — COMMENTED** [review]
Looks good, just one comment.

File: lib/Doctrine/deploy/AddServiceGroupRoles.php (hunk 1 above)
Comment (rowan04) [reply lib/Doctrine/deploy/AddServiceGroupRoles.php:40]: These loops are present in similar files such as https://github.com/GOCDB/gocdb/blob/dev/lib/Doctrine/deploy/AddNGIRoles.php from which I copied them over. Deploying the sample data doesn't work without them, and I think they are there as the corresponding database query above returns an array, but only with one item in it. The foreach loop takes the one data item in the array and makes it a normal string, which is then used later on to make the role in the Role() function.

Comment (gregcorbett) [reply lib/Doctrine/deploy/AddServiceGroupRoles.php:40]: ah okay - I think something like `$roleType = $roleTypes[0]` would be clearer as to what is happening. Could you change that for AddServiceGroupRoles.php?

Comment (rowan04) [reply lib/Doctrine/deploy/AddServiceGroupRoles.php:40]: ok

Comment (rowan04) [reply lib/Doctrine/deploy/AddServiceGroupRoles.php]: done

**Review by gregcorbett — APPROVED** [review]

**Comment by gregcorbett** [comment]
rebasing on top of latest dev

**Pull request merged**